app/models/statistics_announcement.rb

```diff
@@ -38,7 +38,10 @@ def can_publish_to_publishing_api?
   has_many :statistics_announcement_organisations, inverse_of: :statistics_announcement, dependent: :destroy
   has_many :organisations, through: :statistics_announcement_organisations
 
-  validate  :publication_is_matching_type, if: :publication
+  validates_associated :publication, if: :publication,
```

Comment:
Maybe this is overly smart and just having the old (duplicated) validation is more readable.

Whilst this is validation for the publication "field", this is surfaced on the announcement and I feel like the messaging should be flipped somehow. We're not linking to anywhere else on the page as we would for an actual field, so this can be as generic and as "about announcement" as we want.

Reply:
Maybe we can "This announcement has a publication whose type does not match the current selection" -> this validation would only happen when you edit an announcement, as on create you would not have the associated publication yet.

Reply:
Ooh, I've not seen `validates_associated` before 👍 LGTM